.github/workflows/main.yml

```diff
@@ -2,7 +2,6 @@ name: main
 
 on:
   push:
-    branches: [main]
```

Comment:
I believe that this will create double runs CI for contributors who have write access to this repository.
I usually use the following syntax to limit double runs.
```
on:
  push:
    branches: [main]
  pull_request:
    branches: [main]
```

Reply:
@cclauss Thanks for the suggestion. Actually, I want to trigger the CI when I push to my fork in non-main branch so I can run CI before making a pull request. Do you have a suggestion for that setting?

Reply:
@cclauss Thanks for the suggestion. Actually, I want to trigger the CI when I push to my fork in non-main branch so I can run CI before making a pull request. Do you have a suggestion for that setting?

Reply:
OK.  Now I understand the use case.  Your modification will probably work OK for that.